src/handlers/check_commits.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::bail;
+
+use super::Context;
+use crate::{
+    config::Config,
+    github::{Event, IssuesAction},
+};
+
+mod modified_submodule;
+mod non_default_branch;
+
+pub(super) async fn handle(ctx: &Context, event: &Event, config: &Config) -> anyhow::Result<()> {
+    let Event::Issue(event) = event else {
+        return Ok(());
+    };
+
+    if !matches!(event.action, IssuesAction::Opened) || !event.issue.is_pr() {
+        return Ok(());
+    }
+
+    let Some(diff) = event.issue.diff(&ctx.github).await? else {
+        bail!(
+            "expected issue {} to be a PR, but the diff could not be determined",
+            event.issue.number
+        )
+    };
+
+    let mut warnings = Vec::new();
+
+    if let Some(assign_config) = &config.assign {
+        // For legacy reasons the non-default-branch and modifies-submodule warnings
```

Comment:
We might want to eventually introduce a `[check_commits]` config option, but that doesn't need to happen in this PR.

Reply:
Agree, I think it make sense to have a `[check_commits]` config for new warnings.